segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/DebugBookKeeperLogWrapper.java

```diff
@@ -264,6 +274,41 @@ public void deleteDurableLogMetadata() throws DurableDataLogException {
         }
     }
 
+    /**
+     * Deletes all the ledgers of the log starting (including) given starting ledger id.
+     * CAUTION: This is a destructive operation and should be
+     * used wisely for administration purposes (e.g., repair a damaged BookkeeperLog).
+     *
+     * @param startId It refers to the ledger id from where deletion happens.
+     * @return The total number of ledgers deleted.
+     * @throws DurableDataLogException in case there is a problem while deleting ledger.
+     */
+    public int deleteLedgersStartingWithId(long startId) throws DurableDataLogException {
+        LogMetadata metadata = this.log.loadMetadata();
+        List<LedgerMetadata> ledgers = metadata.getLedgers();
```

Comment:
Aren't we updating the metadata after deleting the ledgers?

Reply:
added update metadata after deletion.